src/redux/dragons/dragonsSlice.js

```diff
@@ -1,18 +1,25 @@
 import { createSlice } from '@reduxjs/toolkit';
 
-const initialState = {
-  dragons: [],
-  isLoading: true,
-  error: null,
-};
-
 const dragonsSlice = createSlice({
   name: 'dragons',
-  initialState,
+  initialState: {
+    data: [],
+    isLoading: true,
+    error: null,
+  },
   reducers: {
-
+    setDragonsData: (state, action) => {
+      state.data = action.payload;
+      state.isLoading = false;
+      state.error = null;
+    },
+    setDragonsError: (state, action) => {
+      state.isLoading = false;
+      state.error = action.payload;
+    },
   },
```

Comment:
Since these are connected to the API call, it should be in a fomat of 
```
extraReducers(builder) {
    builder
      .addCase(
// and your code here depending on the API call, like fulfilled, rejected. this way you can use the code you already wrote in a required format.
)
```

Reply:
Hi @hajnaloltyan 
Thank you for reviewing my PR and providing your valuable feedback. I have now implemented the suggested change as per your recommendation. I kindly request you to take another look at the updated code to ensure that it aligns with the desired improvements and meets our project standards.